service/actions/channels.js

```diff
@@ -381,6 +381,11 @@ export function deleteChannel(teamId, channelId) {
             return;
         }
 
+        if (beforeDelete) {
+            const {action, args, _dispatch, _state} = beforeDelete;
```

Comment:
Do we even need that? Why couldn't this just use a thunk since it fires after the request has completed?